### PR TITLE
5846: Fixed the Login,Password field character limit in login page 

### PR DESCRIPTION
**Pull request by pranjaljn97**
Restrict HTML maxlength and size validation on login and password field in   themes/default/login.tmpl.php

### CODE REVIEW

**Comment by cindyli** [comment]
Thanks for the pull request, @pranjaljn97. 

The validation of field lengths should also be verified at the server side before saving into the database for security reason, for example, when requests are not sent through the login page but via other means such as hacking scripts or `curl` commands.

**Comment by pranjaljn97** [comment]
My Pleasure @cindyli.
I have done the server side validation too on the login as well as register functionality.Also added error handling part. Though, password is encrypting on client side only.So, I cant validate length on the server side.
Thank You.
Looking Forward to your Feedback :)

File: themes/default/login.tmpl.php [hunk 1]
```diff
@@ -21,9 +21,15 @@
  * Encrypt login password with sha1
  */
 function encrypt_password() {
-	document.form.form_password_hidden.value = hex_sha1(hex_sha1(document.form.form_password.value) + "<?php echo $_SESSION['token']; ?>");
+	var pass= document.form.form_password.value;
+	if(pass.length>8 && pass.length<15)
```
Comment (cindyli) [inline themes/default/login.tmpl.php]: The password length verification is performed [here](https://github.com/inclusive-design/AChecker/blob/master/jscripts/sha-1factory.js#L219). With numerous password managers out in the market today, it seems quite often that passwords are longer than 15. I'm thinking the proper way is not to limit the length of the passwords, instead, removing this note from the web interface.

The same restriction of "maximum 20 characters for login names" might also make sense to be removed.

Comment (pranjaljn97) [reply themes/default/login.tmpl.php]: I agree We should remove the upper limit on password and login names. 
Though, having some lower limit on cred makes some sense to me.
Should I add or remove anything or you are thinking to close this issue?

Comment (cindyli) [reply themes/default/login.tmpl.php]: This pull request can be adjusted to:
1. On the registration form, there's a note for the `password` field that says: 8 characters minimum, 15 characters maximum. `15 characters maximum` part can be removed;
2. Also remove the note of "20 character maximum" for the `login name` field on the same form;
3. Make sure there isn't any code that checks on removed restrictions.

Comment (pranjaljn97) [reply themes/default/login.tmpl.php]: Okay.

**Pull request closed without merging**